tasks/src/tasks/store.rs

```diff
@@ -0,0 +1,110 @@
+use std::{collections::HashMap, num::NonZeroU32};
+
+use imap_types::{
+    command::CommandBody,
+    core::Vec1,
+    fetch::MessageDataItem,
+    flag::{Flag, StoreResponse, StoreType},
+    response::{Data, StatusBody, StatusKind},
+    sequence::SequenceSet,
+};
+
+use super::TaskError;
+use crate::Task;
+
+#[derive(Clone, Debug)]
+pub struct StoreTask {
+    sequence_set: SequenceSet,
+    kind: StoreType,
+    flags: Vec<Flag<'static>>,
+    uid: bool,
+    output: HashMap<NonZeroU32, Vec1<MessageDataItem<'static>>>,
+}
+
+impl Task for StoreTask {
+    type Output = Result<HashMap<NonZeroU32, Vec1<MessageDataItem<'static>>>, TaskError>;
+
+    fn command_body(&self) -> CommandBody<'static> {
+        CommandBody::Store {
+            sequence_set: self.sequence_set.clone(),
+            kind: self.kind,
+            response: StoreResponse::Answer,
+            flags: self.flags.clone(),
+            uid: self.uid,
+        }
+    }
+
+    fn process_data(&mut self, data: Data<'static>) -> Option<Data<'static>> {
+        if let Data::Fetch { items, seq } = data {
+            self.output.insert(seq, items);
+            None
+        } else {
+            Some(data)
+        }
+    }
+
+    fn process_tagged(self, status_body: StatusBody<'static>) -> Self::Output {
+        match status_body.kind {
+            StatusKind::Ok => Ok(self.output),
```

Comment:
I think we overlooked that before: Here, in the `Status`, there is potentially also a `Code` we want to return. But: This is the case for all `Task`s. Sometimes, the `Code` also makes the `Text` relevant, e.g., `Code::Alert`.

I think we should maybe just ignore this for now. If it is somehow relevant, we can add `StoreTaskV2`.

Reply:
What about making `process_tagged` return a `TaskOutput<'static, Self::Output>` instead:

```rust
pub struct TaskOutput<'a, T> {
    pub output: T,
    pub code: Option<Code<'a>>,
    pub text: Text<'a>,
}
```

Since it impacts all actual PRs, let's wait to merge all of them, and then open a new PR with those changes?

Reply:
Yeah, let's think later :+1: